# Run All

## Import dan Load

Import library yang dibutuhkan

In [176]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings("ignore")

Load data

In [177]:
df_hdi = pd.read_excel('https://drive.google.com/uc?id=1Kl8yCyR7GUcH-hzwiLbcvRK5hHx-ljcJ')

## Data Exploration

Melihat ukuran dataset

In [178]:
df_hdi.shape

(1969, 35)

Melihat 5 data teratas

In [179]:
df_hdi.head()

Country ISO_Code     Level  GDLCODE  ...   2016   2017   2018   2019
0  Afghanistan      AFG  National     AFGt  ...  0.502  0.506  0.509  0.511
1  Afghanistan      AFG    Subnat  AFGr101  ...  0.580  0.585  0.589  0.591
2  Afghanistan      AFG    Subnat  AFGr102  ...  0.493  0.496  0.498  0.501
3  Afghanistan      AFG    Subnat  AFGr103  ...  0.484  0.488  0.491  0.493
4  Afghanistan      AFG    Subnat  AFGr104  ...  0.521  0.525  0.528  0.530

[5 rows x 35 columns]

Melihat info setiap kolom, terutama melihat missing value

In [180]:
df_hdi.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1969 entries, 0 to 1968
Data columns (total 35 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Country   1969 non-null   object 
 1   ISO_Code  1969 non-null   object 
 2   Level     1969 non-null   object 
 3   GDLCODE   1969 non-null   object 
 4   Region    1969 non-null   object 
 5   1990      1607 non-null   float64
 6   1991      1613 non-null   float64
 7   1992      1613 non-null   float64
 8   1993      1613 non-null   float64
 9   1994      1613 non-null   float64
 10  1995      1640 non-null   float64
 11  1996      1640 non-null   float64
 12  1997      1640 non-null   float64
 13  1998      1640 non-null   float64
 14  1999      1676 non-null   float64
 15  2000      1799 non-null   float64
 16  2001      1799 non-null   float64
 17  2002      1813 non-null   float64
 18  2003      1851 non-null   float64
 19  2004      1864 non-null   float64
 20  2005      1919 non-null   floa

## Data Preprocessing

Hanya mengambil data nasional

In [181]:
df_hdi = df_hdi[df_hdi['Level'] == 'National']
df_hdi.shape

(186, 35)

In [182]:
df_hdi.drop(columns=['ISO_Code', 'Level', 'GDLCODE', 'Region'], inplace=True)
df_hdi.head()

Country   1990   1991   1992   1993  ...   2015   2016   2017   2018   2019
0   Afghanistan  0.302  0.308  0.314  0.312  ...  0.499  0.502  0.506  0.509  0.511
9       Albania  0.649  0.631  0.615  0.617  ...  0.787  0.787  0.790  0.793  0.794
22      Algeria  0.573  0.577  0.582  0.586  ...  0.739  0.742  0.746  0.747  0.748
30      Andorra    NaN    NaN    NaN    NaN  ...  0.862  0.866  0.863  0.867  0.868
31       Angola    NaN    NaN    NaN    NaN  ...  0.572  0.578  0.582  0.582  0.582

[5 rows x 31 columns]

Mengganti index menjadi Country

In [183]:
df_hdi.set_index('Country', inplace=True)
df_hdi.head()

1990   1991   1992   1993  ...   2016   2017   2018   2019
Country                                  ...                            
Afghanistan  0.302  0.308  0.314  0.312  ...  0.502  0.506  0.509  0.511
Albania      0.649  0.631  0.615  0.617  ...  0.787  0.790  0.793  0.794
Algeria      0.573  0.577  0.582  0.586  ...  0.742  0.746  0.747  0.748
Andorra        NaN    NaN    NaN    NaN  ...  0.866  0.863  0.867  0.868
Angola         NaN    NaN    NaN    NaN  ...  0.578  0.582  0.582  0.582

[5 rows x 30 columns]

Melakukan transpos

In [184]:
df_hdi = df_hdi.transpose()
df_hdi.head()

Country  Afghanistan  Albania  Algeria  ...  Yemen  Zambia  Zimbabwe
1990           0.302    0.649    0.573  ...  0.400   0.421     0.478
1991           0.308    0.631    0.577  ...  0.401   0.417     0.481
1992           0.314    0.615    0.582  ...  0.404   0.416     0.467
1993           0.312    0.617    0.586  ...  0.407   0.419     0.463
1994           0.307    0.624    0.590  ...  0.409   0.414     0.459

[5 rows x 186 columns]

Mengubah tipe data indeks dari int menjadi str

In [185]:
df_hdi.index = df_hdi.index.map(str)

Melakukan reset index sehingga Year menjadi kolom baru dan melihat tipe data kolom Year yang baru dibuat tersebut

In [186]:
df_hdi = df_hdi.reset_index().rename(columns={'index' : 'Year'})
df_hdi.dtypes['Year']

dtype('O')

Menghilangkan 'Country' dari index

In [187]:
df_hdi.columns.name = None
df_hdi.head()

Year  Afghanistan  Albania  Algeria  ...  Vietnam  Yemen  Zambia  Zimbabwe
0  1990        0.302    0.649    0.573  ...    0.483  0.400   0.421     0.478
1  1991        0.308    0.631    0.577  ...    0.493  0.401   0.417     0.481
2  1992        0.314    0.615    0.582  ...    0.504  0.404   0.416     0.467
3  1993        0.312    0.617    0.586  ...    0.513  0.407   0.419     0.463
4  1994        0.307    0.624    0.590  ...    0.526  0.409   0.414     0.459

[5 rows x 187 columns]

Melihat info dengan banyak float64 ada 186 dan object sebanyak 1

In [188]:
df_hdi.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Columns: 187 entries, Year to Zimbabwe
dtypes: float64(186), object(1)
memory usage: 44.0+ KB


Melakukan Export ke Excel

In [189]:
df_hdi.to_excel(r'HDI_clean.xlsx', index=False, header=True)